swarm/network/stream/snapshot_sync_test.go

```diff
@@ -0,0 +1,568 @@
+	}
+}
+
+/*
+The test generates the given number of chunks,
+then uploads these to a random node.
+Afterwards for every chunk generated, the nearest node addresses
+are identified, syncing is started, and finally we verify
+that the nodes closer to the chunk addresses actually do have
+the chunks in their local stores.
+
+The test loads a snapshot file to construct the swarm network,
+assuming that the snapshot file identifies a healthy
+kademlia network. The snapshot should have 'streamer' in its service list.
+*/
+func runSyncTest(chunkCount int, nodeCount int) error {
+
+	conf = &synctestConfig{}
+	//mapping of nearest node addresses for chunk hashes
+	//nodesToChunksMap = make(map[discover.NodeID][]storage.Key)
+	conf.retrievalMap = make(map[string]map[string]time.Duration)
+	conf.idToAddrMap = make(map[discover.NodeID][]byte)
+	conf.addrToIdMap = make(map[string]discover.NodeID)
+	//First load the snapshot from the file
+	net, err := initNetWithSnapshot(nodeCount)
+	if err != nil {
+		return err
+	}
+	defer net.Shutdown()
+
+	//get the nodes of the network
+	nodes := net.GetNodes()
+	//select one index at random...
+	idx := rand.Intn(len(nodes))
+	//...and get the the node at that index
+	//this is the node selected for upload
+	node := nodes[idx]
+	//iterate over all nodes...
+	for c := 0; c < len(nodes); c++ {
+		//create an array of discovery nodeIDS
+		ids[c] = nodes[c].ID()
+		//and a correspondent array of overlay addresses,
+		//later used for chunk proximity calculation
+		a := network.ToOverlayAddr(ids[c].Bytes())
+		conf.addrs = append(conf.addrs, a)
+		//the proximity calculation is on overlay addr,
+		//the p2p/simulations check func triggers on discover.NodeID,
+		//so we need to know which overlay addr maps to which nodeID
+		conf.idToAddrMap[ids[c]] = a
+		conf.addrToIdMap[string(a)] = ids[c]
+	}
+
+	ppmap = network.NewPeerPot(testMinProxBinSize, ids, conf.addrs)
+	// channel to signal simulation initialisation with action call complete
+	// or node disconnections
+	//disconnectC := make(chan error)
+	//quitC := make(chan struct{})
+
+	//after the test, clean up local stores initialized with createLocalStoreForId
+	defer localStoreCleanup()
+
+	trigger := make(chan discover.NodeID)
+	//triggerCheck defines what will be checked during the test
+	triggerCheck := func(ctx context.Context, id discover.NodeID) (bool, error) {
+		select {
+		case <-ctx.Done():
+			return false, ctx.Err()
+			//case <-disconnectC:
+			//  log.Error("Disconnect event detected")
+			//  return false, ctx.Err()
+		default:
+		}
+
+		log.Debug(fmt.Sprintf("Checking node: %s", id))
+		//select the local store for the given node
+		lstore := stores[id]
+		//if there are more than one chunk, test only succeeds if all expected chunks are found
+		allSuccess := true
+		//this selects which chunks are expected to be found with the given node
+		//localChunks := nodesToChunksMap[id]
+		localChunks := conf.nodesToChunksMap[string(conf.idToAddrMap[id])]
```

Comment:
this should be set only once in the action call

Reply:
It then needs to be a map per node then, as `localChunks` is for one node here, the current node in the `check` function